### ISSUE TITLE
"limit=-1 must be "enabled" with maxLimit=-1" as per #393 not working

### ISSUE THREAD

**Issue by jakobwgnr** [ppetzold/nestjs-paginate#958]
Hey!

I think the changes made in #393 are not fully correct.

SHOULD:
My expectation would be that "maxLimit=-1" would enable the caller to use no pagination if he wants to with "limit=-1".
My expecation would be that the standardfunctionality with e.g. "limit=2" would still work regardless of maxLimit definition

IS:
If I set maxLimit=-1 and limit=2 it will currently give me all entries (hence no pagination) which is not the expected behaviour from my POV.

BR Jakob

**Comment by Helveg** [comment]
Thanks for the report, could you open a PR that adds a failing test case for this? From there, fixing it will be easier :)

**Comment by jakobwgnr** [comment]
@Helveg Please see #959

**Comment by jakobwgnr** [comment]
@Helveg my proposed change would be following:
```    const limit =
        query.limit === PaginationLimit.COUNTER_ONLY
            ? PaginationLimit.COUNTER_ONLY
            : isPaginated === true
            ? maxLimit === PaginationLimit.NO_PAGINATION
                ? query.limit ?? defaultLimit
                : query.limit === PaginationLimit.NO_PAGINATION
                ? defaultLimit
                : Math.min(query.limit ?? defaultLimit, maxLimit)
            : defaultLimit
```
Nevertheless it changes the behaviour of the test "should limit to defaultLimit, if limit is differt FROM NO_PAGINATION ecc...."

But for me it would be the more correct logic to apply the limit if it was given in the query, regardless of the config of "maxLimit"

**Comment by ppetzold** [comment]
yep. definitely also how I had it envisioned :) let's roll with it. breaking changes ftw :) could you add to PR?

**Comment by jakobwgnr** [comment]
@ppetzold @Helveg Change was added to the PR as proposed.

Summary of change:
- Adapted logic to define limit in paginate.ts
- removed a now invalid test case
- added a new test case to cover the new logic
